controlplane/internal/controller/openshiftassistedcontrolplane_controller.go

```diff
@@ -113,33 +114,33 @@ func (r *OpenshiftAssistedControlPlaneReconciler) Reconcile(ctx context.Context,
 		if rerr == nil {
 			patchOpts = append(patchOpts, patch.WithStatusObservedGeneration{})
 		}
-		if err := patchHelper.Patch(ctx, acp, patchOpts...); err != nil {
+		if err := patchHelper.Patch(ctx, oacp, patchOpts...); err != nil {
 			rerr = kerrors.NewAggregate([]error{rerr, err})
 		}
 
 		log.V(logutil.TraceLevel).Info("Finished reconciling OpenshiftAssistedControlPlane")
 	}()
 
-	if acp.DeletionTimestamp != nil {
-		return ctrl.Result{}, r.handleDeletion(ctx, acp)
+	if oacp.DeletionTimestamp != nil {
+		return ctrl.Result{}, r.handleDeletion(ctx, oacp)
 	}
 
-	if !controllerutil.ContainsFinalizer(acp, acpFinalizer) {
-		controllerutil.AddFinalizer(acp, acpFinalizer)
+	if !controllerutil.ContainsFinalizer(oacp, acpFinalizer) {
+		controllerutil.AddFinalizer(oacp, acpFinalizer)
 	}
 
-	acpVersion, err := semver.NewVersion(acp.Spec.Version)
+	acpVersion, err := semver.NewVersion(oacp.Spec.DistributionVersion)
 	if err != nil {
-		log.Error(err, "invalid OpenShift version", "version", acp.Spec.Version)
-		return ctrl.Result{}, err
+		// we accept any format (i.e. latest)
+		log.V(logutil.WarningLevel).Info("invalid OpenShift version", "version", oacp.Spec.DistributionVersion)
```

Comment:
Will having an error here still be ok for the rest of the reconcile function or would it be better to stop reconcile here?

Reply:
If we stop reconcile here, we wouldn't be able to set `latest` as version

Reply:
Ah gotcha, ok